lightning/src/ln/channel.rs

```diff
@@ -1008,6 +1014,22 @@ impl<Signer: Sign> Channel<Signer> {
 		let mut local_htlc_total_msat = 0;
 		let mut value_to_self_msat_offset = 0;
 
+		let mut feerate_per_kw = self.feerate_per_kw;
+		if let Some((feerate, update_state)) = self.pending_update_fee {
+			if self.is_outbound() && generated_by_local {
+				feerate_per_kw = feerate;
+			} else if !self.is_outbound() {
+				if match update_state {
+					// Note that these match the inclusion criteria when scanning
+					// pending_inbound_htlcs below.
+					InboundFeeUpdateState::RemoteAnnounced => !generated_by_local,
+					InboundFeeUpdateState::AwaitingRemoteRevokeToAnnounce => !generated_by_local,
```

Comment:
Not sure I understand the comment or need for the `match` if these have the same value.

Reply:
The point is that the inbound fee update states match exactly the inbound HTLC update states, and the inclusion criteria is identical across them, see the `pending_inbound_htlcs` walk further down in the same function.